File: spec/journey/find_non_domestic_assessor_spec.rb
# frozen_string_literal: true

describe 'Find a Non Domestic Assessor in English', journey: true do
  before do
    visit get_service
    click_on 'Start now'
    find('#label-non-domestic').click
    click_on 'Continue'
    fill_in 'postcode', with: 'SW1A 2AA'
    click_on 'Find'
  end

  it 'shows assessor search results' do
    expect(page).to have_content /assessors in order of distance from SW1A 2AA/
  end
end

describe 'Find a Non Domestic Assessor in Welsh', journey: true do
  before do
    visit get_service_welsh
    click_on 'Dechrau nawr'
    find('#label-non-domestic').click
    click_on 'Parhau'
    fill_in 'postcode', with: 'SW1A 2AA'
    click_on 'Chwiliwch'
  end

  it 'shows assessor search results' do
    expect(page).to have_content /aseswyr yn nhrefn eu pellter o SW1A 2AA/
  end
end

File: cypress/integration/find_non_domestic_assessor_spec.js
describe('Find a non-domestic assessor (English', () => {
  beforeEach(() => {
    cy.visit(Cypress.env(`get_service_en_${Cypress.env('API_STAGE') || 'production'}`))
    cy.contains('Start now').click()
    cy.get('#label-non-domestic').click()
    cy.contains('Continue').click()
    cy.get('input[name=postcode]').type('SW1A 2AA')
    cy.contains('button', 'Find').click()
  })

  it('shows assessor search results', () => {
    cy.get('body').should('contain', 'assessors in order of distance from SW1A 2AA')
  })
})

describe('Find a non-domestic assessor (Welsh)', () => {
  beforeEach(() => {
    cy.visit(Cypress.env(`get_service_cy_${Cypress.env('API_STAGE') || 'production'}`))
    cy.contains('Dechrau nawr').click()
    cy.get('#label-non-domestic').click()
    cy.contains('Parhau').click()
    cy.get('input[name=postcode]').type('SW1A 2AA')
    cy.contains('button', 'Chwiliwch').click()
  })

  it('shows assessor search results', () => {
    cy.get('body').should('contain', 'aseswyr yn nhrefn eu pellter o SW1A 2AA')
  })
})
